modules/basics/src/main/java/com/opengamma/strata/basics/currency/LazyFxRateProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2019 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.basics.currency;
+
+import java.util.function.Supplier;
+
+import com.google.common.base.Suppliers;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * An {@code FxRateProvider} that delays fetching its underlying provider
+ * until actually necessary.
+ * <p>
+ * This is typically useful where you <em>may</em> need a {@code MarketDataFxRateProvider}
+ * but want to delay loading market data to construct the provider until you are sure you actually do need it.
+ */
+public class LazyFxRateProvider implements FxRateProvider {
+
+  private final Supplier<FxRateProvider> target;
+
+  private LazyFxRateProvider(Supplier<FxRateProvider> target) {
```

Comment:
In Strata even private fields/constructors need docs for top level classes, fields are generally Javadoc, but private methods can be `//` such as `// restricted constructor`